### DISCUSSION TITLE
Supabase Studio Self Hosted Multi Tenancy

### DISCUSSION THREAD

**Discussion by henrysachs** [supabase/supabase#16078] [Questions] [answered]
Hi there,

i that last days I tried to self host supabase on a K8s Cluster. There are currently some errors but I managed to get it working. Now I noticed that Components such as Studio currently don't support some kind of multi tenancy in the self hosted version. You also can't create multiple projects in one self hosted instance. 

I would love to have this feature but is it on the roadmap or is it by design not wanted for the self hosted version?

**Comment by GQAdonis** [comment, marked as answer]
From what I have gathered, the only way to do it is to implement your own version of the management API's on the back end and setting an environment variable (NEXT_PUBLIC_IS_PLATFORM) to true.  The Open API spec for the management API to implement is [here](https://supabase.com/docs/reference/api/introduction).  Download that, generate a stub for the server side using whatever language you want via [OpenAPI Generator](https://openapi-generator.tech/), and things should work from there.  Of course, you have to handle the deployment semantics of a "project", etc. yourself (whether by generating Docker VM's, spinning up a cluster, or using an existing cluster to allocate resources separated from other projects).  I have started this work and intend to open source the result, but that is down the list on our roadmap at the moment.
